src/ui/SimulationControls.tsx

```diff
@@ -0,0 +1,145 @@
+import { useContext, useState } from "react";
+import { CoreContext } from "../core/CoreContext";
+import { IAutomatonCore } from "../core/automatonCore";
+import { Kind, Mode } from "../core/core";
+
+export default function VisualWindows() {
+  const coreContext = useContext(CoreContext);
+  // used temporary as for state change signalization
+  const [automatonCore, setAutomatonCore] = useState<IAutomatonCore | null>(null);
+
+  const [simulationWord, setSimulationWord] = useState("aabc");
+  const [simulationWordPos, setSimulationWordPos] = useState(0);
+  const [wordRead, setWordRead] = useState<string[]>([]);
+  const [wordRemaining, setWordRemaining] = useState<string[]>([]);
+  const [steppping, setStepping] = useState(false);
+  const [stepInterval, setStepInterval] = useState<number>(1.0);
+
+  const startSimulation = () => {
+    if (coreContext == undefined) {
+      return;
+    }
+
+    const word = simulationWord.split("");
+
+    const core = coreContext.primary;
+    if (core.kind == Kind.AUTOMATON) {
+      core.runStart(word);
+      setAutomatonCore(core);
+      setSimulationWordPos(0);
+      setWordRemaining(word);
+      setWordRead([]);
```

Comment:
```suggestion
      const e = core.runStart(word);
      if (error !== undefined) {
        alert(error.details);
      } else {
        setAutomatonCore(core);
        setSimulationWordPos(0);
        setWordRemaining(word);
        setWordRead([]);
      }
     
```
tu by malo byť odchytávanie chýb tohto typu, lebo inak ti to spustí simuláciu aj v prípade, že nie si vo visual mode

Reply:
to by bolo sice pekne, ale ta metoda vracia void:

 runStart: (word: string[]) => void;

takze to neviem osetrit

Reply:
my bad, chybala signatura na funkcii, uz odchytavam